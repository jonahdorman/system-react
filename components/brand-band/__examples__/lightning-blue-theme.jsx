import React from 'react';
import BrandBand from '~/components/brand-band'; // `~` is replaced with design-system-react at runtime

class Example extends React.Component {
	render() {
		return (
			<BrandBand
				id="brand-band-lightning-blue"
				className="slds-p-around_small"
				theme="lightning-blue"
			>
				<div className="slds-box slds-theme_default">
					<h3 className="slds-text-heading_label slds-truncate">My App</h3>
				</div>
			</BrandBand>
		);
	}
}

Example.displayName = 'BrandBandExample';

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
