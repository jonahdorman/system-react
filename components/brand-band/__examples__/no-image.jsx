import React from 'react';
import BrandBand from '~/components/brand-band'; // `~` is replaced with design-system-react at runtime

class Example extends React.Component {
	render() {
		return (
			<BrandBand
				id="brand-band-no-image"
				image="none"
				style={{ zIndex: 1 }}
				styleContainer={{ border: '1px solid #eee' }}
			/>
		);
	}
}

Example.displayName = 'BrandBandExample';

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
