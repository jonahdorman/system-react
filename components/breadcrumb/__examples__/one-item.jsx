import React from 'react';
import createReactClass from 'create-react-class';
import IconSettings from '~/components/icon-settings';
import BreadCrumb from '~/components/breadcrumb'; // `~` is replaced with design-system-react at runtime

const Example = createReactClass({
	displayName: 'BreadCrumbExample',

	render () {
		const trail = [
			(<a href="javascript:void(0);">Parent Entity</a>)
		];

		return (
			<IconSettings iconPath="/assets/icons">
				<BreadCrumb trail={trail} />
			</IconSettings>
		);
	}
});

export default Example;	// export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
