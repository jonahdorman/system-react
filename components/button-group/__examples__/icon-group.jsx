import React from 'react';
import createReactClass from 'create-react-class';
import ButtonGroup from '~/components/button-group';
import ButtonStateful from '~/components/button-stateful';
import Dropdown from '~/components/menu-dropdown';
import IconSettings from '~/components/icon-settings';

const Example = createReactClass({
	displayName: 'ButtonGroupExample',

	render() {
		return (
			<IconSettings iconPath="/assets/icons">
				<ButtonGroup>
					<ButtonStateful
						assistiveText={{ icon: 'Show Chart' }}
						buttonVariant="icon"
						iconName="chart"
						iconVariant="border"
						variant="icon"
					/>
					<ButtonStateful
						assistiveText={{ icon: 'Filter List' }}
						iconName="filterList"
						iconVariant="border"
						variant="icon"
					/>
					<Dropdown
						assistiveText={{ icon: 'Settings' }}
						checkmark
						iconCategory="utility"
						iconName="settings"
						iconVariant="more"
						id="icon-dropdown-example"
						onSelect={(item) => {
							console.log(item.label, 'selected');
						}}
						openOn="click"
						options={[
							{ label: 'Bring left panel to front', value: 'A0' },
							{ label: 'Bring right panel to front', value: 'B0' },
						]}
						value="A0"
						variant="icon"
					/>
				</ButtonGroup>
			</IconSettings>
		);
	},
});

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
