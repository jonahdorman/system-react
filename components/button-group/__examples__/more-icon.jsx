import React from 'react';
import createReactClass from 'create-react-class';
import ButtonGroup from '~/components/button-group';
import Button from '~/components/button';
import Dropdown from '~/components/menu-dropdown';
import IconSettings from '~/components/icon-settings';

const Example = createReactClass({
	displayName: 'ButtonGroupExample',

	render() {
		return (
			<IconSettings iconPath="/assets/icons">
				<ButtonGroup>
					<Button label="Refresh" />
					<Button label="Edit" />
					<Button label="Save" />
					<Dropdown
						id="ButtonGroupExampleDropdown"
						assistiveText={{ icon: 'More Options' }}
						buttonVariant="icon"
						iconCategory="utility"
						iconName="down"
						iconVariant="border-filled"
						onSelect={(item) => {
							console.log(item.label, 'selected');
						}}
						openOn="click"
						options={[
							{ label: 'undo', value: 'A0' },
							{ label: 'redo', value: 'B0' },
							{ label: 'activate', value: 'C0' },
						]}
					/>
				</ButtonGroup>
			</IconSettings>
		);
	},
});

export default Example; // export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
