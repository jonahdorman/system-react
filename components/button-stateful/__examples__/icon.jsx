import React from 'react';
import createReactClass from 'create-react-class';
import IconSettings from '~/components/icon-settings';
import ButtonStateful from '~/components/button-stateful'; // `~` is replaced with design-system-react at runtime

const Example = createReactClass({
	displayName: 'ButtonStatefulExample',

	render () {
		return (
			<IconSettings iconPath="/assets/icons">
				<ButtonStateful
					assistiveText="like"
					iconName="like"
					iconSize="large"
					variant="icon"
				/>
			</IconSettings>
		);
	}
});

export default Example;	// export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
