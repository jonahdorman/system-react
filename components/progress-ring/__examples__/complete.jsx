import React from 'react';
import IconSettings from '~/components/icon-settings';
import ProgressRing from '~/components/progress-ring';

class Example extends React.Component {
	render () {
		return (
			<IconSettings iconPath="/assets/icons">
				<div className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center">
					<div className="slds-col--padded">
						<ProgressRing value={100} theme="complete" />
					</div>
					<div className="slds-col--padded">
						<ProgressRing value={100} theme="complete" hasIcon />
					</div>
					<div className="slds-col--padded" />
				</div>
			</IconSettings>
		);
	}
}

Example.displayName = 'ProgressRingDefault';

export default Example;	// export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
