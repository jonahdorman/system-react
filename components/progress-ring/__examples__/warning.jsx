import React from 'react';
import IconSettings from '~/components/icon-settings';
import ProgressRing from '~/components/progress-ring';

class Example extends React.Component {
	render () {
		return (
			<IconSettings iconPath="/assets/icons">
				<div className="slds-grid slds-grid--pull-padded slds-grid--vertical-align-center">
					<div className="slds-col--padded">
						<ProgressRing value={20} theme="warning" />
					</div>
					<div className="slds-col--padded">
						<ProgressRing value={20} theme="warning" hasIcon />
					</div>
					<div className="slds-col--padded">
						<ProgressRing value={100} theme="warning" />
					</div>
				</div>
			</IconSettings>
		);
	}
}

Example.displayName = 'ProgressRingDefault';

export default Example;	// export is replaced with `ReactDOM.render(<Example />, mountNode);` at runtime
