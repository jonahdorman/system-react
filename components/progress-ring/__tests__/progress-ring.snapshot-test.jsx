/* eslint-env jest */
import React from 'react';
import { renderMarkup } from '../../../tests/snapshot-helpers';
import IconSettings from '../../../components/icon-settings';
import * as ProgressRing from '../__examples__/examples';

Object.keys(ProgressRing).forEach((name) => {
	const wrapper = (element) => () => (<IconSettings iconPath="/assets/icons">{element()}</IconSettings>);

	test(`${name} DOM Snapshot`, () => {
		expect(renderMarkup(wrapper(ProgressRing[name]))).toMatchSnapshot();
	});
});

