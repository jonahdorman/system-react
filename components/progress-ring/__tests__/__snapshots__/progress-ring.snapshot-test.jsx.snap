// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BASE_0 DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"0\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 1 0 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`BASE_100 DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`BASE_PARTIAL DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`COMPLETE_100 DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_complete\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`COMPLETE_WITH_ICON DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_complete\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"><span title=\\"complete\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#check\\"></use></svg></span></div>
</div>"
`;

exports[`EXPIRED_100 DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_expired\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`EXPIRED_PARTIAL DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_expired\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`EXPIRED_WITH_CUSTOM_ICON DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_expired\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"><span><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#lock\\"></use></svg></span></div>
</div>"
`;

exports[`EXPIRED_WITH_ICON DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_expired\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"><span title=\\"Expired\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#error\\"></use></svg></span></div>
</div>"
`;

exports[`WARNING_100 DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_warning\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"100\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 1 1 1 -2.4492935982947064e-16 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`WARNING_PARTIAL DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_warning\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"></div>
</div>"
`;

exports[`WARNING_WITH_ICON DOM Snapshot 1`] = `
"<div class=\\"slds-progress-ring slds-progress-ring_warning\\">
  <div class=\\"slds-progress-ring__progress\\" role=\\"progressbar\\" aria-valuemin=\\"0\\" aria-valuemax=\\"100\\" aria-valuenow=\\"20\\"><svg viewBox=\\"-1 -1 2 2\\"><path class=\\"slds-progress-ring__path\\" d=\\"M 1 0 A 1 1 0 0 1 0.30901699437494745 0.9510565162951535 L 0 0\\"></path></svg></div>
  <div class=\\"slds-progress-ring__content\\"><span title=\\"Warning\\"><svg aria-hidden=\\"true\\" class=\\"slds-icon slds-icon-text-default\\"><use xlink:href=\\"/assets/icons/utility-sprite/svg/symbols.svg#warning\\"></use></svg></span></div>
</div>"
`;
